src/server/native_controllers/spawn_native_controller.js

```diff
@@ -0,0 +1,48 @@
+import { fork } from 'child_process';
+import { transformDeprecations, Config } from '../config';
+
+async function defaultConfig(settings) {
+  return await Config.withDefaultSchema(
```

Comment:
Hmm, but this still doesn't support config from plugins, right?

Reply:
Yeah, when the [config service is initially extended](https://github.com/elastic/kibana/blob/master/src/server/config/config.js#L30-L32) it only includes values for the keys of the default schema... Maybe we can default to `_.get(settings, key)`? Defaults from the plugin schema won't be applied but maybe that's the responsibility of the native controller to handle. I'm still a little concerned about the fact that the config could totally include secure values like passwords but be written in plain text to `/proc/<pid>/cmdline`

Reply:
What if this didn't wait for the start event but was delivered as soon as the config started and said it was ready to receive the config?

Reply:
I agree that this isn't an ideal scenario, and if we were to pass "secrets" via this extension point it would "leak" them. It also doesn't allow for us to pass plugin specific configuration over. However, using IPC (including start) to pass over this specific configuration if it's used to influence where the native controller spawns the next process isn't feasible, because the entire point of the system call filters is to prevent arbitrary process spawning.

Reply:
> However, using IPC (including start) to pass over this specific configuration if it's used to influence where the native controller spawns the next process isn't feasible, because the entire point of the system call filters is to prevent arbitrary process spawning.

Couldn't we send a single setup message before we activate the sandbox, make sure that the native controller only allows a single setup message, and then move on with initialization?